Source/Core/Common/BitField2.h

```diff
@@ -0,0 +1,544 @@
+#pragma once
```

Comment:
Previous BitField.h was BSD licensed. Would you like to license this as BSD as well so other projects can integrate this more easily, or do you prefer keeping it as GPL?

Reply:
I actually have been wondering about this lately.  If I were to contribute wholly original code to Dolphin Emulator, what rights do I retain?  Am I able to license it under a more permissive license than GPL in one of my own repositories, but still give Dolphin Emulator full permission to use it?  I was beginning to cook up a better MetroWerks Linker Map reader before this PR spiraled into the crazy thing it has now become.  Software licensing is something I'm still really unfamiliar with, so I'd appreciate any guidance.

Reply:
In some projects in the past where I pretended to understand the licensing, I would use my online handle instead of my real name for the copyright info.  I'm not going to pretend I have perfect anonymity online, but I prefer to use my pseudonym.  Do you have any advice for this as well?

Reply:
MIT and Unlicense are the only licenses I really am familiar with.  I'm happy with whatever, really, so long as it doesn't somehow get locked up in legal limbo.

Reply:
(The usual disclaimer applies for this whole comment: I'm not a lawyer, and this is not legal advice)

>  If I were to contribute wholly original code to Dolphin Emulator, what rights do I retain

You retain all rights to the code you wrote yourself. Licensing just gives others permission to do things with your code, such as copying and modifying it.

>  Am I able to license it under a more permissive license than GPL in one of my own repositories, but still give Dolphin Emulator full permission to use it?

Yes, you can publish the same code (assuming you wrote all of it yourself) at a different place under a different license. The license you provide via the Dolphin source code is not affected by this.

BitField.h's license header is a bit funky in this regard. It's licensed as BSD (as mentioned starting from line 4), but then relicensed as GPL since that's what Dolphin as a whole uses. In hindsight I'd probably omit the explicit GPL notice, but the idea is that if people want to use BitField.h in other projects, they can do so without worrying about Dolphin being GPL-licensed as a whole.

> In some projects in the past where I pretended to understand the licensing, I would use my online handle instead of my real name for the copyright info. I'm not going to pretend I have perfect anonymity online, but I prefer to use my pseudonym. Do you have any advice for this as well?

People more active in the project than I should comment on this, though I'll note that using a pseudonym makes it difficult for project owners to verify your contribution doesn't legally poison the repository (such as, for example, submitting code that actually belongs to your employer).

Reply:
In my opinion (also not a lawyer), it'd be fine to use your pseudonym or even just "Dolphin emulator project".

`BitSet.h` also has a different license, but for it all that's listed is `SPDX-License-Identifier: CC0-1.0` ([Creative Commons Public Domain Declaration](https://creativecommons.org/share-your-work/public-domain/cc0)), which functions similarly to the Unlicense (e.g. both do not have an attribution requirement, so no name is listed).  For that file that's the only license listed, which is probably preferable.